io.github.ferraridamiano.ConverterNOW.yaml

```diff
@@ -0,0 +1,45 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/flatpak/flatpak-builder/master/data/flatpak-manifest.schema.json
+app-id: io.github.ferraridamiano.ConverterNOW
+runtime: org.freedesktop.Platform
+runtime-version: "21.08"
+sdk: org.freedesktop.Sdk
+command: converternow
+separate-locales: false
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --share=network
+  - --share=ipc
+  - --device=dri
+modules:
+  - name: converternow
+    buildsystem: simple
+    build-commands:
+      - cp -r ConverterNOW /app/
+      - chmod +x /app/ConverterNOW/converternow
+      - mkdir -p /app/bin
+      - ln -s /app/ConverterNOW/converternow /app/bin/converternow
+      - install -Dm644 -t "${FLATPAK_DEST}/share/applications/" io.github.ferraridamiano.ConverterNOW.desktop
+      - install -Dm644 icon16.png "${FLATPAK_DEST}/share/icons/hicolor/16x16/apps/${FLATPAK_ID}.png"
+      - install -Dm644 icon32.png "${FLATPAK_DEST}/share/icons/hicolor/32x32/apps/${FLATPAK_ID}.png"
+      - install -Dm644 icon48.png "${FLATPAK_DEST}/share/icons/hicolor/48x48/apps/${FLATPAK_ID}.png"
+      - install -Dm644 icon128.png "${FLATPAK_DEST}/share/icons/hicolor/128x128/apps/${FLATPAK_ID}.png"
+#      - install -Dm644 -t "${FLATPAK_DEST}/share/appdata/" io.github.ferraridamiano.ConverterNOW.desktop
+    sources:
+      - type: archive
+        url: https://github.com/ferraridamiano/ConverterNOW/releases/download/v3.1.1/converternow-linux.tar.gz
+        dest: ConverterNOW
+        sha256: cf00cca5a35ae6decbfb5fb5490fbf98c2af48bd1a845a7f9f18c8fae9f977e4
+      - type: file
+        path: io.github.ferraridamiano.ConverterNOW.appdata.xml
+      - type: file
+        path: io.github.ferraridamiano.ConverterNOW.desktop
```

Comment:
there is a .desktop in the upstream repository, you should use it instead of copying it here.

same for the icon.

Reply:
Why do you say that I'm not using the .desktop file?

In the `build-commands` there is `install -Dm644 -t "${FLATPAK_DEST}/share/applications/" io.github.ferraridamiano.ConverterNOW.desktop`

Sorry for all these mistakes

Reply:
no I mean the one that is in the upstream repository.

Reply:
The icon is now retrieved from the project repository. But for the `.desktop file` it could be a problem: currently I'm using snapcraft CI/CD and I'm not sure I can reuse that file since the icon line should be something like: `Icon=${SNAP}/meta/gui/icon.png` which is incompatible with snapcraft. Alternatively I could import that file and replace the icon line in the `build-commands`, but i'm not sure wheter it is a good practice or not.

Reply:
1. you can change that with `desktop-file-edit`
2. This `Icon` line is invalid (according to the spec) because the icon value is just the name (no extension)

it is always a better practice to patch that to duplicate.